omnibus/config/software/python3.rb

```diff
@@ -64,25 +65,66 @@
     block do
       FileUtils.rm_f(Dir.glob("#{install_dir}/embedded/lib/python#{major}.#{minor}/distutils/command/wininst-*.exe"))
     end
-  end
+  else
+    dependency "vc_redist_14"
+
+    vcrt140_root = "#{Omnibus::Config.source_dir()}/vc_redist_140/expanded"
+    ###############################
+    # Setup openssl dependency... #
+    ###############################
+    mkdir "externals/openssl-bin-3.0.15/amd64/include"
+    # Copy the import library to have them point at our own built versions, regardless of
+    # their names in usual python builds
+    copy "#{install_dir}/embedded/lib/libcrypto.dll.a", "externals/openssl-bin-3.0.15/amd64/libcrypto.lib"
+    copy "#{install_dir}/embedded/lib/libssl.dll.a", "externals/openssl-bin-3.0.15/amd64/libssl.lib"
+    copy "#{install_dir}/embedded/lib/libssl.dll.a", "externals/openssl-bin-3.0.15/amd64/libssl.lib"
+    # Copy the actual DLLs, be sure to keep the same name since that's what the IMPLIBs expect
+    copy "#{install_dir}/embedded/bin/libssl-3-x64.dll", "externals/openssl-bin-3.0.15/amd64/libssl-3.dll"
+    command "touch externals/openssl-bin-3.0.15/amd64/libssl-3.pdb"
+    copy "#{install_dir}/embedded/bin/libcrypto-3-x64.dll", "externals/openssl-bin-3.0.15/amd64/libcrypto-3.dll"
+    command "touch externals/openssl-bin-3.0.15/amd64/libcrypto-3.pdb"
+    # The applink "header"
+    copy "#{install_dir}/embedded/include/openssl/applink.c", "externals/openssl-bin-3.0.15/amd64/include/"
+    # And finally the headers:
+    copy "#{install_dir}/embedded/include/openssl", "externals/openssl-bin-3.0.15/amd64/include/"
+    # Now build python itself
+    # -e to enable external libraries. They won't be fetched if already
+    # present, but the modules will be built nonetheless.
+
+    ###############################
+    # Build Python...             #
+    ###############################
+    command "PCbuild\\build.bat -e --pgo"
+    command "dir PCbuild/amd64/"
+    # Install the build artifact to their expected locations
+    copy "PCbuild/amd64/*.exe", "#{windows_safe_path(python_3_embedded)}/"
+    copy "PCbuild/amd64/*.dll", "#{windows_safe_path(python_3_embedded)}/"
+    mkdir "#{windows_safe_path(python_3_embedded)}/DLLs"
+    copy "PCbuild/amd64/*.pyd", "#{windows_safe_path(python_3_embedded)}/DLLs/"
+    mkdir "#{windows_safe_path(python_3_embedded)}/libs"
+    copy "PCbuild/amd64/*.lib", "#{windows_safe_path(python_3_embedded)}/libs"
+    copy "Lib", "#{windows_safe_path(python_3_embedded)}/"
+    command "copy /y \"#{windows_safe_path(vcrt140_root)}\\*.dll\" \"#{windows_safe_path(python_3_embedded)}\""
 
-else
-  dependency "vc_redist_14"
 
-  # note that starting with 3.7.3 on Windows, the zip should be created without the built-in pip
-  source :url => "https://dd-agent-omnibus.s3.amazonaws.com/python-windows-#{version}-amd64.zip",
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏Do we build the exact same versions we previously had? Asking for the changelog mainly

Reply:
Yes, we still target 3.12.6 for now. The main change is that we can now use any openssl version we want, not the one that is shipped with the python release